frontend/src/pages/NewRunV2.tsx

```diff
@@ -116,23 +158,35 @@ function NewRunV2(props: NewRunV2Props) {
     { enabled: !!apiPipelineVersion, staleTime: Infinity },
   );
 
+  const templateString =
+    templateStringFromRunId !== undefined ? templateStringFromRunId : templateStringFromPipelineId;
```

Comment:
NIT: Prefer `===` over `!==`. You can swap the `templateStringFromRunId` and `templateStringFromPipelineId` to improve readability.